app/web/features/auth/jail/PasswordSection.tsx

```diff
@@ -0,0 +1,116 @@
+import { TextField, Typography, useMediaQuery } from "@material-ui/core";
+import { Alert } from "@material-ui/lab";
+import Button from "components/Button";
+import { Empty } from "google-protobuf/google/protobuf/empty_pb";
+import { Error as GrpcError } from "grpc-web";
+import { useTranslation } from "next-i18next";
+import { useForm } from "react-hook-form";
+import { useMutation } from "react-query";
+import { service } from "service";
+import { theme } from "theme";
+import useChangeDetailsFormStyles from "features/auth/useChangeDetailsFormStyles";
+
+interface SetPasswordVariables {
+  password?: string;
+}
+
+interface SetPasswordFormData extends SetPasswordVariables {
+  passwordConfirmation?: string;
+}
```

Comment:
```suggestion
interface SetPasswordVariables {
  password: string;
}

interface SetPasswordFormData extends SetPasswordVariables {
  passwordConfirmation: string;
}
```
I think they should both be required since we aren't letting the user sending through empty passwords in the validations. You've probably copied from the other form and I think it was only optional because we were allowing people to clear their password before :p

That'd fix the other type error on `service.jail.setPassword(...)` complaining about `string | undefined` is not assignable to `string`

Reply:
Thanks!